pkg/component/observability/opentelemetry/operator/operator.go

```diff
@@ -350,6 +350,9 @@ func (o *openTelemetryOperator) deployment() *appsv1.Deployment {
 									},
 								},
 							},
+							SecurityContext: &corev1.SecurityContext{
+								AllowPrivilegeEscalation: ptr.To(false),
+							},
```

Comment:
Cosmetic nit: could we move after the `Resources` so that we are consistent with the PRs that @georgibaltiev made and we follow the alphabetic order?